argus/shell.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2016 Cloudbase Solutions Srl
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from __future__ import print_function
+
+import argparse
+import os
+import subprocess
+import sys
+import tempfile
+
+from argus.backends.tempest import manager
+from argus.config import ci
+from argus import exceptions
+
+import requests
+import six
+from six.moves import urllib_parse as urlparse
+from os_testr import subunit2html
+
+
+def _download_resource(url, location):
+    """Download a file from a remote url.
+
+    :param url: The URL that points to the resource
+    :param location: Where to save the resource
+    """
+    response = requests.get(url)
+    if response.status_code != 200:
+        raise exceptions.ArgusEnvironmentError("Download failed, %s .", url)
+    with open(location, 'wb') as file_handle:
+        file_handle.write(response.text)
+
+
+def download_argus_resource(resource_path, location):
+    """Download an argus Resource.
+
+    :param resouce_path: Path of the resource relative to the
+                         argus `resources` directory
+    :param location: Where to save the resource
+    """
+    base_url = ci.RESOURCES_LINK
+    base_url = base_url.rsplit("/", 1)[0]
+
+    url_resource = urlparse.urljoin(base_url,
+                                    resource_path)
+    _download_resource(url_resource, location)
+
+
+def _get_images(image_id, image_name):
+    """Return a list with all the id's."""
+    images_list = []
+    if image_id:
+        try:
+            mng = manager.APIManager()
```

Comment:
this all images features needs to be re discussed, as we need a flavor mapping, too